server.go

```diff
@@ -31,12 +33,26 @@ type quoteResponse struct {
 	cyrptokey string
 }
 
+var (
+	delayVal = rand.Intn(4) + 1
+)
+
 func main() {
 	// If we don't provide a seed for rand then it hehaves as if
 	// we ran Seed(1). It's not safe to run this in concurrent
 	// code so I'm doing it here!
+	flag.Parse()
 	rand.Seed(time.Now().Unix())
 
+	if len(flag.Args()) != 0 {
+		parsedVal, err := strconv.Atoi(flag.Args()[0])
```

Comment:
Nit: Could use `flag.Int` or `flag.VarInt` to handle this when `flag.Parse()` is called.